File: proveedor-service/src/main/java/com/tingeso/proveedorservice/services/ProveedorService.java
package com.tingeso.proveedorservice.services;

import com.tingeso.proveedorservice.entities.ProveedorEntity;
import com.tingeso.proveedorservice.repositories.ProveedorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
@Service
public class ProveedorService {
    @Autowired
    ProveedorRepository proveedorRepository;

    public ArrayList<ProveedorEntity> obtenerProveedores(){
        return (ArrayList<ProveedorEntity>) proveedorRepository.findAll();
    }

    public void guardarProveedor(ProveedorEntity proveedor){
        proveedorRepository.save(proveedor);
    }

    public void borrarTodo(){
        proveedorRepository.deleteAll();
    }
}

File: frontend/src/services/ProveedorService.js
import axios from 'axios';

const PROVEEDOR_API_URL = "http://localhost:8080/proveedor";

class ProveedorService {
    
    obtenerProveedores(){
        return axios.get(PROVEEDOR_API_URL);
    }

    crearProveedor(proveedor){
        return axios.post(PROVEEDOR_API_URL,proveedor);
    }

    borrarTodo(){
        return axios.post(PROVEEDOR_API_URL + "/borrarTodo");
    }
}

export default new ProveedorService()